apis/src/main/java/xyz/bookself/controllers/user/UserController.java

```diff
@@ -35,8 +35,43 @@ public UserController(UserRepository repository) {
     }
 
     @PostMapping(value = "/new-user", consumes = {MediaType.APPLICATION_JSON_VALUE})
-    public ResponseEntity<User>createNewUser(@RequestBody User newUser){
+    public ResponseEntity<User>createNewUser(@RequestBody NewUserDTO newUserDTO){
+        User newUser = new User();
+        newUser.setUsername(newUserDTO.getUsername());
+        newUser.setEmail(newUserDTO.getEmail());
+        newUser.setPasswordHash(newUserDTO.getPasswordHash());
         newUser.setCreated(LocalDate.now());
         return new ResponseEntity<>(userRepository.save(newUser), HttpStatus.OK);
     }
+}
+
+class NewUserDTO
+{ 
+    private String username;
+    private String passwordHash;
```

Comment:
Might make sense to rename this to just password since we're going to receive the plain-text password and not the hash.

Reply:
Okay. I'll do a quick change.